include/rrc/io/IOBuffer.h

```diff
@@ -0,0 +1,38 @@
+/**
+ *  @autor Nikita Somenkov
+ *  @date 08/10/2016
+ */
+#pragma once
+
+#include <memory>
+
+namespace rrc {
+
+    class IOBuffer
```

Comment:
Maybe you should reconsider this class. It looks like std::string with some limitations. Or maybe you should use Copy On Write here. 
And of course it would be nice to have some iterators, begin, end etc. Constructor that accepts begin and end would be great too. And don't forget about move constructor.